Mathlib/SetTheory/Cardinal/Ordinal.lean

```diff
@@ -1445,3 +1447,69 @@ theorem extend_function_of_lt {α β : Type _} {s : Set α} (f : s ↪ β) (hs :
 -- end Bit
 
 end Cardinal
+
+section Initial
+
+namespace Ordinal
+
+open Cardinal
+
+/--
+`initial i` (notation: `ω_ i`) gives the first ordinal of cardinality
+`aleph i`. Thus `ω_ 0 = ω` and `ω_ 1` is the first uncountable ordinal.
+You can also write `ω_ 1` as `ω₁`.
+-/
+def initial (x : Ordinal.{u}) : Ordinal.{u} := (aleph x).ord
+
+@[inherit_doc]
+scoped notation "ω_" => Ordinal.initial
+@[inherit_doc]
+scoped notation "ω₁" => Ordinal.initial 1
+
+@[simp]
+theorem card_initial (i : Ordinal): card (initial i) = aleph i := card_ord _
+
+/--
+The first (infinite) initial ordinal is `ω`.
+-/
+@[simp]
+lemma initial0 : ω_ 0 = ω := by
+  unfold initial
+  simp
+
+/--
+An initial ordinal is a limit ordinal.
+-/
+lemma isLimit_initial (i : Ordinal): (ω_ i).IsLimit := ord_isLimit (aleph0_le_aleph i)
+
+lemma omega_lt_omega1 : ω < ω₁ := ord_aleph0.symm.trans_lt (ord_lt_ord.mpr (aleph0_lt_aleph_one))
+
+end Ordinal
+
+end Initial
+
+section OrdinalIndices
+/-!
+### Cardinal operations with ordinal indices
+
+Results on cardinality of ordinal-indexed families of sets.
+-/
+namespace Cardinal
+
+open scoped Cardinal
+
+/--
+Bounding the cardinal of an ordinal-indexed union of sets.
+-/
+lemma mk_iUnion_Ordinal_le_of_le {β : Type _} {κ : Cardinal} {i : Ordinal}
+    (hi : i ≤ κ.ord) (hκ : ℵ₀ ≤ κ) (A : Ordinal → Set β)
+    (hA : ∀ j < i, #(A j) ≤ κ) :
+    #(⋃ j < i, A j) ≤ κ := by
+  simp_rw [← mem_Iio, biUnion_eq_iUnion, iUnion, iSup, ← i.enumIsoOut.symm.surjective.range_comp]
+  apply ((mk_iUnion_le _).trans _).trans_eq (mul_eq_self hκ)
+  rw [mk_ordinal_out]
+  exact mul_le_mul' (card_le_of_le_ord hi) <| ciSup_le' <| (hA _ <| typein_lt_self ·)
```

Comment:
Variable names here haven't been changed; do you agree it's better to use `c` and `o` here following the rest of the file?

Reply:
I'm actually following your suggestions without hesitations---i.e., I agree completely. 

I just added all the proposed commits in one batch and only later I realized that it did not pass CI. (I did not--and do not--know how to make a more comprehensive commit later and add you as a coauthor.)
Also, I realized I didn't remove the `awaiting-review` label before I double check the stuff. 

Btw, did you see Mario's [comments](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236404.20defining.20initial.20ordinals/near/382512001)? I was a bit concerned but perhaps we can go ahead with this.

Reply:
> I'm actually following your suggestions without hesitations---i.e., I agree completely.
> 
> I just added all the proposed commits in one batch and only later I realized that it did not pass CI. (I did not--and do not--know how to make a more comprehensive commit later and add you as a coauthor.) 

My plan was to complete the variable renaming and indent styling after committing the first batch.

Reply:
> I just added all the proposed commits in one batch and only later I realized that it did not pass CI.

No worries!

> I did not--and do not--know how to make a more comprehensive commit later and add you as a coauthor.

Normally you'd just commit yourself and add me in a Co-authored-by line in the PR description.

> Btw, did you see Mario's [comments](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236404.20defining.20initial.20ordinals/near/382512001)? I was a bit concerned but perhaps we can go ahead with this.

I actually couldn't find where the range of `Cardinal.ord` is used to define initial ordinals; do you know?

Reply:
> Normally you'd just commit yourself and add me in a Co-authored-by line in the PR description.
>

Done
 
> > Btw, did you see Mario's [comments](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236404.20defining.20initial.20ordinals/near/382512001)? I was a bit concerned but perhaps we can go ahead with this.
> 
> I actually couldn't find where the range of `Cardinal.ord` is used to define initial ordinals; do you know?

No, I don't, and I guess it's not. I think he meant that adding this particular def was not necessary, or that adding it as an extra layer would be problematic from an API point of view

Reply:
> Normally you'd just commit yourself and add me in a Co-authored-by line in the PR description.

No longer to do this since you already commited my suggestions (and I also commited). I've removed the line because otherwise my name will appear twice.